Electricity Generation Through Enhanced Geothermal Systems

LukasFrankenQ opened the pull request:
**Addresses Issue** PyPSA/pypsa-eur#541.

This PR adds potential for electricity generation through _enhanced geothermal systems_ (EGS) as in [this paper](https://www.sciencedirect.com/science/article/abs/pii/S0306261920312551#:~:text=In%20the%20EGS%20process%2C%20high,or%20a%20binary%20power%20plant.). The data's resolution is on the level of countries.

EGS draws heat from great depths (1-10 km) to achieve the temperatures of 100-150 C required for electricity generation through steam turbines or binary cycle power plants.

The EGS heat can also be used for (duh) heating. This is currently not implemented, as we are still in contact with the authors of the paper to obtain the relevant cost estimations. However, in terms of implementation both electricity and heat would be quite canonical, and in that sense I thought it sensible to move forward with the electricity part and create a new PR for heat at a later stage.

**Open questions (that probably should find good answers before merging):**

1. Currently, the egs-scripts draw from `egs_costs.xlsx` (20kb) and `egs_global_potential.xlsx` (500kb), which are simply stored in `data/`. Is this okay or should they be added to the databundle?
2. How EGS electricity changes the system is still under investigation. Its costs are assumed to be subject to a strong learning rate, and a naive (and temporally sparse) run with 2050 costs builds egs capacity up to the posed constraints in _Germany_, _Austria_, _Switzerland_ and parts of _Italy_. I am currently setting up the infrastructure (i.e. the Edinburgh cluster) to test the implications more rigorously through experiments with multiple investment periods. Results will be summarized in a short report.
3. The implementation uses `pycountry`, which is a tremendously slim package, which I would love to add to the `environment.yaml`. I suppose I would make a PR at `pypsa/pypsa-eur` for this?

**Possible Improvements:**

1. Heat becomes available through drilling at various depths. Through the cost of drilling, potential becomes a function of LCOE, and the paper shares its data thus as: _x MW available at LCOE <= 5 Euro/MWh, y MW available at LCOE of 5-10 Euro/MWh, etc..._. The present implementation coarsens this binning, to the LCOE steps 0-50, 50-100, and 100-150 Euro/MWh. To accommodate the linear nature of the model, each of the bins is implemented as a separate generator sitting on the same bus. Our freedom with regards to choice of bins is limited, as it is dictated by the data provided through mail exchange with the paper authors. However, I am happy to discuss how we can obtain capital and marginal cost from LCOE, to make our own choice of binning, it should be possible while respecting the different discount rates etc..

2. The implementation of heat and electricity could potentially be coupled, since EGS can lend itself to generation through CHP. Landau is such an example, where hot water from the well at 160 C is used for electricity generation and the residual heat with temperature 70-80 C for district heating. See [here](https://www.sciencedirect.com/science/article/abs/pii/S0306261920312551#:~:text=In%20the%20EGS%20process%2C%20high,or%20a%20binary%20power%20plant.), [here](https://www.globalccsinstitute.com/archive/hub/publications/121288/geothermal-electricity-combined-heat-power.pdf), and [here](https://reader.elsevier.com/reader/sd/pii/S1364032117310341?token=1F0864FD7F51BEAB4999E2ECFE1992856704655E949DF030B076CF85269AA9C01D4DB4F7BA69AD54024174A3F0AAE533&originRegion=eu-west-1&originCreation=20230207134427).
    
Big thanks to @lisazeyen for tips along the way!

LukasFrankenQ commented:
Also, I do not know why github thinks I made a whole new Snakefile...

lisazeyen commented:
@LukasFrankenQ thanks a lot for your PR! 
We should add the additional data to the data bundle and not the the github repository. Can you convert the excel to an csv and change the code accordingly?

fneum committed:
limit openpyxl version

LukasFrankenQ committed:
temp changes

LukasFrankenQ commented on config.default.yaml:
```diff
@@ -281,7 +280,9 @@ sector:
   biosng: false
-
+  egs: true
+  egs_lifetime: 30 # estimate from 'From hot rock to useful energy:...' !!Currently at odds with pypsa-eur/data/costs.csv!!
```
Tops, done! :)

LukasFrankenQ commented on scripts/prepare_sector_network.py:
```diff
@@ -2771,6 +2770,78 @@ def set_temporal_aggregation(n, opts, solver_name):
+    )
+
+    # should be done in analogy to marginal costs of wind/solar etc.
+    marginal_cost = pd.Series(
```
Fixed!

LukasFrankenQ commented on scripts/prepare_sector_network.py:
```diff
@@ -2771,6 +2770,78 @@ def set_temporal_aggregation(n, opts, solver_name):
+        suffix=f" egs_lcoe_{cutoff}",
+        bus=buses,
+        carrier="enhanced geothermal",
+        p_nom=0,
```
Fixed!

LukasFrankenQ commented on scripts/prepare_sector_network.py:
```diff
@@ -2811,6 +2884,23 @@ def set_temporal_aggregation(n, opts, solver_name):
 
+    if options["egs"]:
+
+        # should probably be set much earlier in the workflow
```
Alright!

LukasFrankenQ commented on scripts/prepare_sector_network.py:
```diff
@@ -2771,6 +2770,78 @@ def set_temporal_aggregation(n, opts, solver_name):
+    # scripts/build_egs_potential.py 
+    # for a discussion of the following steps
+
+    if dr > 0:
```
This might be a silly question, but is discount rate usually assumed uniform across technologies or does every tech have its own?

LukasFrankenQ commented:
> @LukasFrankenQ thanks a lot for your PR! We should add the additional data to the data bundle and not the the github repository. Can you convert the excel to an csv and change the code accordingly?

Absolutely! Should I send them over to be added on zenodo?

LukasFrankenQ committed:
addressed (so far some) comments from PR review

LukasFrankenQ committed:
emphasized year vs costs_year

LukasFrankenQ commented on scripts/prepare_sector_network.py:
```diff
@@ -2811,6 +2884,23 @@ def set_temporal_aggregation(n, opts, solver_name):
+              )
+
+        logger.info("Adding Enhanced Geothermal Potential")
+        year = snakemake.config["costs"]["year"]
```
I adjusted the variable name to emphasize that this refers to the cost year

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/PyCQA/docformatter: v1.6.0.rc1 → v1.5.1](https://github.com/PyCQA/docformatter/compare/v1.6.0.rc1...v1.5.1)
- [github.com/macisamuele/language-formatters-pre-commit-hooks: v2.6.0 → v2.7.0](https://github.com/macisamuele/language-formatters-pre-commit-hooks/compare/v2.6.0...v2.7.0)

fneum committed:
Merge pull request #477 from PyPSA/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

LukasFrankenQ committed:
added sustainable potential

FabianHofmann committed:
replace progressbar by tqdm

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
ci: flush python terminal output

FabianHofmann committed:
helpers: undo position fixing in tqdm

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
ci: revert python buffer

FabianHofmann committed:
build_renewable_profiles: remove progressbar import

fneum committed:
Merge pull request #478 from PyPSA/tqdm-display

replace progressbar by tqdm

fneum committed:
cluster option 'm' only aggregates conventional carriers not excluded

fneum committed:
Merge branch 'master' into cluster-m

fneum committed:
ci: reduce duplications of test/config.test1.yaml

fneum committed:
fix path to test config

fneum committed:
allow disabling max_depth through config overwrite

LukasFrankenQ committed:
CO2 intensity now taken from technology-data

LukasFrankenQ committed:
allow geothermal to contribute to district heating

LukasFrankenQ committed:
changed egs potential back to non-sustainable

LukasFrankenQ committed:
bug fixes and minor cleanup

LukasFrankenQ committed:
fixed a new bug in the generation of p_nom_max

LukasFrankenQ committed:
cleanup

fneum committed:
allow false for max_depth

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
Merge pull request #479 from PyPSA/cluster-m

cluster option 'm' only aggregates conventional carriers not excluded

fneum committed:
Merge branch 'master' into ci-config

LukasFrankenQ committed:
solved merge conflicts

FabianHofmann committed:
config: separate solver_options
solve_network: adjust code to new solver_options config

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
update release notes

fneum committed:
Merge pull request #480 from PyPSA/ci-config

ci: reduce duplications of test/config.test1.yaml

fneum committed:
Merge remote-tracking branch 'pypsa-eur-sec/master' into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
split solve network rules, take summary scripts from Sec

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
merge docs

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
trigger readthedocs webhook

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
trigger readthedocs webhook II

fneum committed:
conform to pypsa-eur logging level definition

fneum committed:
how to handle upcoming releases

fneum committed:
doc: reorganise pre-merge PyPSA-Eur release notes

fneum committed:
remove superfluous license file

fneum committed:
use line length factor from single config source

fneum committed:
remove unneeded currency conversion USD to EUR

fneum committed:
align cost fill values to pypsa-eur implementation

fneum committed:
add input file handling of pypsa-eur-sec rules using cutout

fneum committed:
remove duplicate retrieve_cost_data

fneum committed:
COSTS are data not resources

fneum committed:
fix path references in Snakefile

FabianHofmann committed:
ci modularization fix cutout retrieval and profile in dag

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
Merge branch 'master' into ci-config

FabianHofmann committed:
Merge pull request #485 from PyPSA/ci-config

Ci config fixes

FabianHofmann committed:
Merge branch 'master' into merge-pypsa-eur-sec

fneum committed:
correct aggregation strategies

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
add missing cutout input

FabianHofmann committed:
scripts/ remove summary_dir and results_dir references

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
correct benchmark

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
fix all lv references to ll, pickup right location of default lifetime

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
prepare CI for all three cases

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
fix solve_all_elec_networks

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
remove plot_p_nom_max

fneum committed:
re-allow 'all' clusters wilcard value

fneum committed:
technology-data not dependent on run

fneum committed:
fix logs and benchmark directories

fneum committed:
remove figsize from config

FabianHofmann committed:
fix cutout retrieval in ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
delete unnecessary log files
fix make_summary for myopic

fneum committed:
disable windows/macos to reduce CI load

fneum committed:
add spdx identifiers to scripts

fneum committed:
use mock_snakemake from pypsa-eur

fneum committed:
remove accidental log additions

fneum committed:
use progress_retrieve from single source

fneum committed:
remove unused function in _helpers

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
merge helper.py into _helpers.py

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
reuse compatibility for pypsa-eur-sec code

fneum committed:
snakefmt to Snakefile

fneum committed:
doc: update landing page

LukasFrankenQ committed:
towards implementation of geothermal as chp

LukasFrankenQ committed:
minor cleanup

LukasFrankenQ committed:
corrections enhanced geothermal -> geothermal heat

LukasFrankenQ committed:
removed geothermal from spatial

FabianHofmann committed:
move snakemake object out of imap process functions

FabianHofmann committed:
ci: reactivate windows and mac

FabianHofmann committed:
build industrial production: fix partial function in imap

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
snakefile: fix benchmark path

FabianHofmann committed:
build industrial demand/production: ensure compat with multiprocesses on win/mac

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LukasFrankenQ committed:
fixed a bug in capital cost assignment

FabianHofmann committed:
ci overnight/myopic: use glpk

FabianHofmann committed:
scripts: apply linting fixes

FabianHofmann committed:
snakefile: introduce RESOURCES, RESULTS, BENCHMARKS, LOGS paths

FabianHofmann committed:
Snakefile: replace else statements by if statement to satisfy snakefmt

FabianHofmann committed:
reenable snakefmt

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
allow to disable progressbar

LukasFrankenQ committed:
temp changes

fneum committed:
modularise Snakefile using rules/*.smk files

fneum committed:
add conda directive to all rules for use with --use-conda

fneum committed:
correct paths to scripts in .smk files

fneum committed:
apply snakefmt to rules/

fneum committed:
make cutouts protected
https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html\#protected-and-temporary-files

fneum committed:
data/bundle files may be ancient

fneum committed:
retrieval rules get retries directive for added robustness

fneum committed:
add a purge rule

fneum committed:
add rule for dag, use shutil to purge outputs

fneum committed:
add message for dag rule

fneum committed:
Snakefile: correct indent

fneum committed:
add graphviz to environment

fneum committed:
add logfiles to all rules

fneum committed:
snakefmt to Snakefile

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
new minimum snakemake version is 7.7

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
upload resources and results as artifacts

FabianHofmann committed:
ci: update actions/checkout

FabianHofmann committed:
ci: reset data cache number

LukasFrankenQ committed:
temp changes

fneum committed:
eliminate manual country code conversion, adjust mock_snakemake to ll wildcard

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
sector-coupled: run on subset of selected countries

fneum committed:
sort countries

fneum committed:
Merge branch 'merge-pypsa-eur-sec' into sec-subset-countries

fneum committed:
speed-up CI, just a week rather than a month, 4 nodes, 40H

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
keep CI running if one OS fails

fneum committed:
Merge branch 'sec-subset-countries' of github.com:PyPSA/pypsa-eur into sec-subset-countries

fneum committed:
Merge pull request #623 from PyPSA/sec-subset-countries

Run also sector-coupled build part for subset of countries

fneum committed:
ci: add --rerun-triggers=mtime to improve caching

fneum committed:
Merge branch 'sec-subset-countries' into merge-pypsa-eur-sec

fneum committed:
ci: revert to 5 regions

fneum committed:
add minimal description of all scripts

FabianHofmann committed:
helpers: allow to overwrite config in mock_snakemake and allow mock_snakemake to be called from repo root

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
docs: rearrange sections

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pz-max committed:
add Linopy to PyPSA

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

FabianHofmann committed:
harmonive solve_network across workflow

LukasFrankenQ committed:
functional implementation of geothermal CHP

FabianHofmann committed:
fix Nyears scaling

FabianHofmann committed:
prepare_sector: revert taking mean
config.overnight: revert hourly resolution

FabianHofmann committed:
adress review comments

FabianHofmann committed:
Update scripts/solve_network.py

Co-authored-by: Fabian Neumann <fabian.neumann@outlook.de>

FabianHofmann committed:
solve_network: reinsert docstring

FabianHofmann committed:
solve_network: add line breaks in docstring

FabianHofmann committed:
follow up: fix img path

FabianHofmann committed:
Merge pull request #625 from PyPSA/linopy-eur

add Linopy to PyPSA-Eur

FabianHofmann committed:
build_sequestration: fix url with docformatter

FabianHofmann committed:
solve_network: check if sector_opts is in wildcard

FabianHofmann committed:
follow up

LukasFrankenQ committed:
adjusted cost of district heating part to literature

LukasFrankenQ committed:
bug fix in the geothermal chp constraints

LukasFrankenQ committed:
bug fix in geothermal chp constraints

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/codespell-project/codespell: v2.2.2 → v2.2.4](https://github.com/codespell-project/codespell/compare/v2.2.2...v2.2.4)
- [github.com/snakemake/snakefmt: v0.8.1 → v0.8.2](https://github.com/snakemake/snakefmt/compare/v0.8.1...v0.8.2)

LukasFrankenQ committed:
added population layout to compute egs share of dh

fneum committed:
Merge pull request #627 from PyPSA/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

FabianHofmann committed:
codespell: fix typo & add word to exception list

LukasFrankenQ committed:
towards inclusion of part of area feasible for dh from egs

LukasFrankenQ committed:
fixed a bug with dh assignment in egs

fneum committed:
enhance documentation, with instructions for sector-coupled / electricity-only studies

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
remove manual country codes

fneum committed:
docs: add release notes and document new config options

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
docs: add missing configuration options

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
docs: cartopy cannot be installed from pip alone

fneum committed:
docs: complete instructions for sector-coupling tutorial2

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

lisazeyen committed:
config.default.yaml

set default share_cutout: True

lisazeyen committed:
add more sector_opts wildcards

fneum committed:
docs: limit matplotlib<3.6

fneum committed:
docs: limit matplotlib>3.5.1,<3.6

lisazeyen committed:
correct snakemake call for sector coupled networks

fneum committed:
encapsulate imports in helpers to minimise requirements for docs

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
doc: fix number of columns for sector-oprt

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
doc: add ways to cite PyPSA-Eur

fneum committed:
doc: add list of publications

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LukasFrankenQ committed:
solved merhe conflicts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

fneum committed:
Merge branch 'merge-pypsa-eur-sec' of github.com:PyPSA/pypsa-eur into merge-pypsa-eur-sec

fneum committed:
fix license specification of doc/publications.bib

fneum committed:
doc: implement review feedback

fneum committed:
doc: implement review feedback (amend)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LukasFrankenQ committed:
Updated description, now uses vresutils for annuity

LukasFrankenQ committed:
solved merge conflicts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LukasFrankenQ commented:
Just a small update, as I am removing the "Draft" tag.

In short, based on the feedback, and on conversations with experts on EGS, I made the respective changes.
The current state is a naive implementation of this paper: https://doi.org/10.1016/j.apenergy.2020.115769

This paper makes very optimistic estimations of future cost of EGS, so runs of years ~2040-2050 would build EGS extensively (on the magnitude of wind and solar), the only reason they don't is because the respective co2 constraints are not commensurate with estimated EGS emissions of 120gCO2/kWh_el. Hence, the current state is in line with what Tom and I agree to be sensible: EGS is in the model, is however not built extensively in default runs. 

In that sense, and as far as I am concerned, the PR could be merged - after another probable round of corrections of course. However, towards a research article, I will further improve the implementation, and add some degrees of freedom. These might clutter the model more than anything else, so I think their inclusion in the main repo is optional.

They also feel slightly removed from the spirit of PyPSA-Eur-Sec, as they build on vague research estimations, and not on more robust numbers from the DAE: These quantities themselves are what is being investigated, hence they don't lend themselves to inclusion in the model.

This is a non-exhaustive list of them:

1. A risk factor, that simulates the fact the deep geothermal project are not guaranteed to work out.
2. The learning rate that leads to low costs of EGS in later cost years is 7.5%, I would like to test different numbers.
3. A learning rate specific to countries.
4. A constraint on the area feasible for EGS: Only close to electric infrastructure.
5. The degree to which produced heat is used for dh has an effect on the efficiency of electricity generation, this should be captured in some way.

Let me know if some of them are also interesting for the default model, I can of course add them.

All the best,
Lukas

fneum committed:
Merge pull request #484 from PyPSA/merge-pypsa-eur-sec

Merge pypsa eur sec

fneum committed:
add release notes for v0.8.0

fneum committed:
add release notes for v0.8.0

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/macisamuele/language-formatters-pre-commit-hooks: v2.7.0 → v2.8.0](https://github.com/macisamuele/language-formatters-pre-commit-hooks/compare/v2.7.0...v2.8.0)
- [github.com/snakemake/snakefmt: v0.8.2 → v0.8.3](https://github.com/snakemake/snakefmt/compare/v0.8.2...v0.8.3)

fneum committed:
Merge branch 'master' of github.com:PyPSA/pypsa-eur

fneum committed:
Merge pull request #629 from PyPSA/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

aodenweller committed:
Bugfix CPLEX settings

aodenweller committed:
Merge branch 'master' of github.com:PyPSA/pypsa-eur into bugfix_cplex

aodenweller committed:
Update release note with CPLEX bugfix

fneum committed:
Merge pull request #630 from aodenweller/bugfix_cplex

Bugfix typo in CPLEX configuration settings

LukasFrankenQ committed:
remove kwarg unit from egs generator

LukasFrankenQ committed:
Merge branch 'master' of https://github.com/LukasFrankenQ/pypsa-eur-sec

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.1.0 → 23.3.0](https://github.com/psf/black/compare/23.1.0...23.3.0)

fneum committed:
Merge pull request #636 from PyPSA/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

LukasFrankenQ committed:
merged into pypsa-eur

LukasFrankenQ committed:
lit review of egs, removed later

LukasFrankenQ committed:
started adding flexible EGS

LukasFrankenQ committed:
temp save of egs paper notes

LukasFrankenQ committed:
started buiding egs constraints

LukasFrankenQ committed:
added egs constraint from faults and got DH shares

LukasFrankenQ committed:
added temperature-dependent capacity factor

LukasFrankenQ committed:
added flexible geothermal

LukasFrankenQ committed:
adjusted mock snakemake

LukasFrankenQ committed:
egs spatial constraints made functional

LukasFrankenQ committed:
added more granular egs potential data

LukasFrankenQ committed:
added granular egs data to prepare_sector script

LukasFrankenQ committed:
adjusted constraints to new data

LukasFrankenQ committed:
added more performant run option

LukasFrankenQ committed:
added temp plotting scripts

LukasFrankenQ committed:
added performance increases in EGS

LukasFrankenQ committed:
prepared from sweep over egs for dh costs

LukasFrankenQ committed:
refactored geothermal plotting methods

LukasFrankenQ committed:
added sweep pipeline

LukasFrankenQ committed:
added multiple links for egs for district heating

LukasFrankenQ committed:
removed egs requirement from prepare sector rule
